src/app/services/graph-client/MsalService.ts

```diff
@@ -0,0 +1,90 @@
+import { AuthenticationParameters } from 'msal/lib-commonjs/AuthenticationParameters';
+import { DEFAULT_USER_SCOPES } from '../graph-constants';
+import { msalApplication } from './MsalAgent';
+
+const loginType = getLoginType();
+const defaultUserScopes = DEFAULT_USER_SCOPES.split(' ');
+
+export async function logIn(): Promise<any> {
+  const loginRequest: AuthenticationParameters = {
+    scopes: defaultUserScopes,
+  };
+
+  if (loginType === 'POPUP') {
+    try {
+      await msalApplication.loginPopup(loginRequest);
+      const response = await msalApplication.acquireTokenPopup(loginRequest);
+      return response.accessToken;
+    } catch (error) {
+      return null;
+    }
+  } else if (loginType === 'REDIRECT') {
+    await msalApplication.acquireTokenRedirect(loginRequest);
```

Comment:
Looks like this could throw as well. 
Also, since we are not really using redirects, does it sound fair to remove similar sections from the code?

Reply:
It does make sense